MediaAdmin/EventSubscriber/MediaCreatedSubscriber.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace OpenOrchestra\MediaAdmin\EventSubscriber;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use OpenOrchestra\MediaAdmin\FileAlternatives\FileAlternativesManager;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use OpenOrchestra\MediaAdmin\MediaEvents;
+use OpenOrchestra\MediaAdmin\Event\MediaEvent;
+
+/**
+ * Class MediaCreatedSubscriber
+ */
+class MediaCreatedSubscriber implements EventSubscriberInterface
+{
+    public $medias = array();
+    protected $fileAlternativesManager;
+    protected $objectManager;
+
+    /**
+     * @param fileAlternativesManager $fileAlternativesManager
+     * @param ObjectManager           $objectManager
+     */
+    public function __construct(FileAlternativesManager $fileAlternativesManager, ObjectManager $objectManager)
+    {
+        $this->fileAlternativesManager = $fileAlternativesManager;
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @param MediaEvent $event
+     */
+    public function addMedia(MediaEvent $event)
+    {
+        $media = $event->getMedia();
+        $this->medias[] = $media;
+    }
+
+    /**
+     * Generate images
+     */
+    public function generateAlternatives()
+    {
+        /** @var MediaInterface $media */
+        foreach ($this->medias as $media) {
+            $media = $this->fileAlternativesManager->generateThumbnail($media);
```

Comment:
@nicolasThal : I think you're speaking of the methods rather than the managers. If so, both methods return the media object because they can alter it. For instance the generateThumbnail method injects the thumbnail name in the media object. The generateAlternatives method could inject the alternative names too. That way the manager can update its local media and flush it at the end of the process (line 52).

An alternative could be to give to the strategy a reference rather than the object but we don't do this elsewhere, plus I'm not a big fan of that way of doing.